tests/python/test_f16.py

```diff
@@ -218,3 +218,60 @@ def paint(t: float):
             pixels[i, j] = 1 - iterations * 0.02
 
     paint(0.03)
+
+
+@ti.test(arch=archs_support_f16)
+def test_atomic_add_f16():
+    f = ti.field(dtype=ti.f16, shape=(2))
+
+    @ti.kernel
+    def foo():
+        # Parallel sum
+        for i in range(1000):
+            f[0] += 1.12
+
+        # Serial sum
+        for _ in range(1):
+            for i in range(1000):
+                f[1] = f[1] + 1.12
+
+    foo()
+    assert (f[0] == f[1])
```

Comment:
This can fail, we need an epsilon when comparing floats as its not commutative

Reply:
Thanks!